File: frontend/src/types/terminal.ts
export enum TerminalStatus {
  ACTIVE = 'ACTIVE',
  INACTIVE = 'INACTIVE',
  SUSPENDED = 'SUSPENDED',
  TERMINATED = 'TERMINATED'
}

export enum TerminalType {
  CAT = 'CAT',
  POS = 'POS',
  MOBILE = 'MOBILE',
  KIOSK = 'KIOSK',
  ONLINE = 'ONLINE'
}

export const TERMINAL_STATUS_LABELS: Record<TerminalStatus, string> = {
  [TerminalStatus.ACTIVE]: '정상',
  [TerminalStatus.INACTIVE]: '비활성',
  [TerminalStatus.SUSPENDED]: '정지',
  [TerminalStatus.TERMINATED]: '해지'
};

export const TERMINAL_TYPE_LABELS: Record<TerminalType, string> = {
  [TerminalType.CAT]: 'CAT 단말기',
  [TerminalType.POS]: 'POS 단말기',
  [TerminalType.MOBILE]: '모바일',
  [TerminalType.KIOSK]: '키오스크',
  [TerminalType.ONLINE]: '온라인'
};

export interface TerminalDto {
  id: string;
  tid: string;
  catId?: string;
  terminalType: TerminalType;
  merchantId: string;
  merchantName?: string;
  merchantCode?: string;
  organizationId?: string;
  organizationName?: string;
  serialNumber?: string;
  model?: string;
  manufacturer?: string;
  installAddress?: string;
  installDate?: string;
  status: TerminalStatus;
  lastTransactionAt?: string;
  config?: Record<string, unknown>;
  createdAt: string;
  updatedAt: string;
}

export interface TerminalListParams {
  page?: number;
  size?: number;
  status?: TerminalStatus;
  terminalType?: TerminalType;
  merchantId?: string;
  organizationId?: string;
  search?: string;
}

export interface TerminalCreateRequest {
  tid: string;
  catId?: string;
  terminalType: TerminalType;
  merchantId: string;
  serialNumber?: string;
  model?: string;
  manufacturer?: string;
  installAddress?: string;
  installDate?: string;
  config?: Record<string, unknown>;
}

export interface TerminalUpdateRequest {
  catId?: string;
  serialNumber?: string;
  model?: string;
  manufacturer?: string;
  installAddress?: string;
  status?: TerminalStatus;
  config?: Record<string, unknown>;
}

File: src/main/java/com/korpay/billpay/domain/entity/Terminal.java
package com.korpay.billpay.domain.entity;

import com.korpay.billpay.domain.enums.TerminalStatus;
import com.korpay.billpay.domain.enums.TerminalType;
import io.hypersistence.utils.hibernate.type.json.JsonBinaryType;
import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.Type;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.LastModifiedDate;
import org.springframework.data.jpa.domain.support.AuditingEntityListener;

import java.time.LocalDate;
import java.time.OffsetDateTime;
import java.util.Map;
import java.util.UUID;

@Entity
@Table(name = "terminals")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
@EntityListeners(AuditingEntityListener.class)
public class Terminal {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @Column(name = "tid", nullable = false, unique = true, length = 50)
    private String tid;

    @Column(name = "cat_id", length = 50)
    private String catId;

    @Enumerated(EnumType.STRING)
    @Column(name = "terminal_type", nullable = false, length = 20)
    private TerminalType terminalType;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "merchant_id", nullable = false)
    private Merchant merchant;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "organization_id")
    private Organization organization;

    @Column(name = "serial_number", length = 100)
    private String serialNumber;

    @Column(name = "model", length = 100)
    private String model;

    @Column(name = "manufacturer", length = 100)
    private String manufacturer;

    @Column(name = "install_address", columnDefinition = "TEXT")
    private String installAddress;

    @Column(name = "install_date")
    private LocalDate installDate;

    @Enumerated(EnumType.STRING)
    @Column(name = "status", nullable = false, length = 20)
    @Builder.Default
    private TerminalStatus status = TerminalStatus.ACTIVE;

    @Type(JsonBinaryType.class)
    @Column(name = "config", columnDefinition = "jsonb")
    private Map<String, Object> config;

    @Column(name = "last_transaction_at")
    private OffsetDateTime lastTransactionAt;

    @CreatedDate
    @Column(name = "created_at", nullable = false, updatable = false)
    private OffsetDateTime createdAt;

    @LastModifiedDate
    @Column(name = "updated_at", nullable = false)
    private OffsetDateTime updatedAt;

    @Column(name = "deleted_at")
    private OffsetDateTime deletedAt;
}
